signature-aggregator/api/api.go

```diff
@@ -0,0 +1,155 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package api
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ava-labs/awm-relayer/signature-aggregator/aggregator"
+	"github.com/ava-labs/awm-relayer/types"
+	"github.com/ava-labs/awm-relayer/utils"
+	"go.uber.org/zap"
+)
+
+const (
+	RawMessageAPIPath = "/aggregate-signatures/by-raw-message"
+	defaultQuorumNum  = 67
+)
+
+// Defines a request interface for signature aggregation for a raw unsigned message.
+// Currently a copy of the `ManualWarpMessageRequest` struct in relay_message.go
```

Comment:
Can you add a TODO to combine these? We can probably have `awm-relayer` consume this package directly for manual Warp messages. We may want to add filtering for only supported subnets in that case though.

Reply:
I'm just removing that comment for now since it's lying. Right now manual warp messages require specifying source address which is not part of this interface. I'm not convinced that we need that though, but either way that would be a breaking change on the relayer side.

Reply:
Cool, sounds good